tasks/swarm_join.rb

```diff
@@ -6,11 +6,11 @@
 require 'puppet'
 
 def swarm_join(advertise_addr, listen_addr, token, manager_ip)
-  cmd_string = 'docker swarm join'
-  cmd_string += " --advertise-addr=#{advertise_addr}" unless advertise_addr.nil?
-  cmd_string += " --listen-addr=#{listen_addr}" unless listen_addr.nil?
-  cmd_string += " --token=#{token}" unless token.nil?
-  cmd_string += " #{manager_ip}" unless manager_ip.nil?
+  cmd_string = ['docker', 'swarm', 'join']
+  cmd_string += [" --advertise-addr=#{advertise_addr}"] unless advertise_addr.nil?
+  cmd_string += [" --listen-addr=#{listen_addr}"] unless listen_addr.nil?
+  cmd_string += [" --token=#{token}"] unless token.nil?
+  cmd_string += [" #{manager_ip}"] unless manager_ip.nil?
```

Comment:
We should try to avoid using += here.

I believe [Array#concat](https://ruby-doc.org/core-2.1.1/Array.html#method-i-concat) would be preferable in this instance.

Also lets rename the variable to cmd rather than cmd_string.

Reply:
See above.

Reply:
In this case you have changed from concatenating a string to an array so we should generally lean towards the more idiomatic approach.